src/main/java/zean/Parser.java

```diff
@@ -0,0 +1,176 @@
+            }
+
+            String deadline = input.substring(indexOfBy + 3).strip();
+            if (deadline.isBlank()) {
+                throw new DukeException("OOPS!!! You forgot to specify the deadline.");
+            }
+            if (!DATE_PATTERN.matcher(deadline).matches()) {
+                throw new DukeException("Hmm, I don't understand the date. "
+                        + "Use this format: YYYY-MM-DD");
+            }
+
+            output = tasks.add(description, deadline);
+            break;
+        case "event":
+            int indexOfFrom = input.indexOf("/from");
+            int indexOfTo = input.indexOf("/to");
+            if (indexOfFrom == -1) {
+                throw new DukeException("OOPS!!! You forgot to specify the starting date."
+                        + "\nUse \"/from\" to do so.");
+            }
+            if (indexOfTo == -1) {
+                throw new DukeException("OOPS!!! You forgot to specify the ending date."
+                        + "\nUse \"/to\" to do so.");
+            }
+
+            description = input.substring(5, indexOfFrom);
+            if (description.isBlank()) {
+                throw new DukeException("Hmm, the description of an event cannot be empty :(");
+            }
+            String from = input.substring(indexOfFrom + 5, indexOfTo).strip();
+
+            if (from.isBlank()) {
+                throw new DukeException("OOPS!!! You forgot to specify the starting date.");
+            }
+
+            String to = input.substring(indexOfTo + 3).strip();
+            if (to.isBlank()) {
+                throw new DukeException("OOPS!!! You forgot to specify the ending date.");
+            }
+
+            if (!DATE_PATTERN.matcher(from).matches() || !DATE_PATTERN.matcher(to).matches()) {
+                throw new DukeException("Hmm, I don't understand the date. "
+                        + "Use this format: YYYY-MM-DD");
+            }
+            output = tasks.add(description, from, to);
+            break;
+        case "find":
+            description = input.substring(4);
+            System.out.println(description);
+            if (description.isBlank()) {
+                throw new DukeException("Please provide the keyword for me to search.");
+            }
+            output = tasks.find(description);
+            break;
+        default:
+            throw new DukeException("OOPS!!! I'm sorry, but I don't understand what that means :-(");
+        }
+        return output;
+    }
+
+    /**
+     * Parses the formatted text read from the file and adds the created task to the ArrayList.
+     *
+     * @param arrList The ArrayList to contain the tasks.
+     * @param text The formatted text read from the file.
+     * @return The number of tasks added successfully to the ArrayList.
+     */
+    public static int parseToTask(ArrayList<Task> arrList, String text) {
```

Comment:
instead of adding an `ArrayList` directly, you might consider passing in a `TaskList` instead.